[RFR] Updated bz coverage utility to not update bugs in coverage= marker

prichard77 opened the pull request:
## Purpose or Intent


- __Updating test utility__ miq bz coverage so that it no longer updates the qe_test_coverage flag in bugzilla for bugs in coverage= markers within tests. Also replacing log messages with exceptions when bugzilla credentials are missing. 


### PRT Run

Suggestions on what to use here?

prichard77 commented:
These updates were made quickly with getting the utility working so we could update bugs being the priority. I wanted to get it out for review comments. I am trying to determine where we want to catch the Exceptions thrown if we have no bz creds, and what specifically to do when caught. Thinking we want to write something to store.terminalreporter and end the test, but from where?

And I want to make sure, when I removed the coverage= code , I didn't create an issue with something else using the code I updated. Seems unlikely, but I have no idea.

john-dupuy commented on cfme/fixtures/bzs.py:
```diff
@@ -23,16 +23,12 @@ def pytest_addoption(parser):
+    store.terminalreporter.write("Loading automated BZs ...\n", bold=True)
     bz_list = []
     for item in items:
-        if "automates" not in item._metadata and "coverage" not in item._metadata:
-            continue
```
@prichard77 rather than removing the handling of the `coverage`/manual cases altogether, I think we should make it an option (`--include-manual`) that defaults to `False`. 

Something like
```   
group.addoption(
        '--include-manual',
        action='store_true',
        default=False,
        dest='include_manual',
        help='Include manual coverage in the report'
    )
```

john-dupuy replied:
Then the `miq bz` command can also have an `--include-manual` flag

prichard77 replied:
Sounds like a great idea. working on it...

john-dupuy replied:
OH that's right, we should just use that! So we can just add an `--include-manual` flag in `miq bz coverage` and then include it in the `pytest.main` call based on that.

john-dupuy replied:
That way no changes are required to `cfme.fixtures.bzs.py`

prichard77 commented:
I removed my initial changes. Now only scripting.py has been updated (still have to work on checking for creds). I added an option to coverage. I have done a little testing on this, but it's tough to test thoroughly as we have already updated all of the Bugs to reflect the correct/current state. There is no work to be done by the utility now.

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -44,18 +44,19 @@
 
 
-def get_report(directory):
+def get_report(directory, inc_manual=True):
```
@prichard77 since this is dependent on the bugzilla credentials being correct, and this is the source method for all the data that the rest of the module is using, I think we should check here whether or not credentials are available.

The instance given by `cfme.utils.bz.Bugzilla.from_config` will have either the `username` or the `api_key` attribute set if credentials were provided.

If the credentials are wrong, that a whole other issue that I don't think you need to handle here.

prichard77 replied:
Updated as suggested.

mshriver replied:
Good catch too on the instance attribute being `key`, even though the kwarg is `api_key`

john-dupuy commented on cfme/scripting/bz.py:
```diff
@@ -224,6 +225,15 @@ def list(directory, bz_status):
 @click.argument("directory", default="cfme/tests")
+@click.option(
+    "-m",
+    "--inc_man",
```
@prichard77 for consistency with the `pytest` arg, I suggest we just call this `--include-manual`

prichard77 replied:
Done

prichard77 commented:
I was going to make sure I added --inc_manual to all of the functions where it was applicable, but it looks like all the others are fine like they are. All others are specifically looking for 'coverage' so turning it off would make no sense. Please let me know if I missed one for which the option should apply.

john-dupuy commented on cfme/scripting/bz.py:
```diff
@@ -224,6 +231,15 @@ def list(directory, bz_status):
 @click.argument("directory", default="cfme/tests")
+@click.option(
+    "-m",
+    "--inc_manual",
```
```suggestion,
    "--include-manual",
```

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -263,8 +279,8 @@ def list(directory, bz_status):
 )
-def coverage(directory, set_bzs, bz_status):
-    info = get_report(directory)
+def coverage(directory, inc_manual, set_bzs, bz_status):
```
```suggestion
def coverage(directory, include_manual, set_bzs, bz_status):
```

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -263,8 +279,8 @@ def list(directory, bz_status):
-def coverage(directory, set_bzs, bz_status):
-    info = get_report(directory)
+def coverage(directory, inc_manual, set_bzs, bz_status):
+    info = get_report(directory, inc_manual=inc_manual)
```
```suggestion
    info = get_report(directory, include_manual=include_manual)
```

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -44,18 +45,24 @@
 
 
-def get_report(directory):
+def get_report(directory, inc_manual=True):
```
```suggestion
def get_report(directory, include_manual=True):
```

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -44,18 +45,24 @@
-    ])
+        "--generate-bz-report"
+               ]
+    if inc_manual:
```
```suggestion
    if include_manual:
```

mshriver commented on cfme/scripting/bz.py:
```diff
@@ -224,6 +231,15 @@ def list(directory, bz_status):
+@click.option(
+    "-m",
+    "--include-manual",
+    "inc_manual",
```
```suggestion
    "include_manual",
```
